### PR TITLE
[UI smoke test] For #6531: Adds new UI test for themes

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Also moved tests for default browser and locale change to this new class for a better structure.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt [hunk 1]
```diff
@@ -103,6 +122,24 @@ class SwitchLocaleTest {
...
 
+    @SmokeTest
+    @Test
+    fun changeDefaultBrowserTest() {
+        homeScreen {
+        }.openMainMenu {
+        }.openSettings {
+        }.openGeneralSettingsMenu {
+            clickSetDefaultBrowser()
+            if (Build.VERSION.SDK_INT <= 23) {
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt:133]: Let's switch this to compare against `Build.VERSION_CODES.M` https://apilevels.com

File: app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt [hunk 2]
```diff
@@ -111,11 +148,22 @@ class SwitchLocaleTest {
             val res = context.applicationContext.resources
             val config = res.configuration
             config.setLocale(Locale(locale!!))
-            if (VERSION.SDK_INT >= 25) {
+            if (Build.VERSION.SDK_INT >= 25) {
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt:151]: and `Build.VERSION_CODES.N` here

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt (hunk 1 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt]: done

File: app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt (hunk 2 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/focus/activity/SettingsGeneralTest.kt]: done

**Review by AaronMT — APPROVED** [review]

**Pull request merged**